torch/distributed/algorithms/ddp_comm_hooks/default_hooks.py

```diff
@@ -57,11 +66,36 @@ def fp16_compress_hook(
         compressed_tensor, group=group_to_use, async_op=True
     ).get_future()
 
-    def decompress(fut):
-        decompressed_tensor = bucket.get_tensors()[0]
-        # Decompress in place to reduce the peak memory.
-        # See: https://github.com/pytorch/pytorch/issues/45968
-        decompressed_tensor.copy_(fut.value()[0].div_(world_size))
-        return [decompressed_tensor]
+    decompress_fp16 = decompress_fp16_callable(bucket, world_size)
+    return fut.then(decompress_fp16)
+
+def fp16_compress_wrapper(hook: callable) -> callable:
```

Comment:
The output should be `torch.futures.Future`.

Reply:
In the example:

```    
>>> state = PowerSGDState(process_group=process_group, matrix_approximation_rank=1, start_powerSGD_iter=10)
    >>> ddp_model.register_comm_hook(state, fp16_compress_wrapper(powerSGD_hook))
```
We have 

```
    def register_comm_hook(self, state: object, hook: callable):
```

so I am pretty sure the wrapper needs to return a callable, whereas the callable needs to return a `torch.futures.Future`. Let me know if I'm missing something!

Reply:
Yeah, you are right :)